skema/text_reading/text_reading/src/main/scala/org/ml4ai/skema/text_reading/CosmosTextReadingPipeline.scala

```diff
@@ -0,0 +1,157 @@
+package org.ml4ai.skema.text_reading
+
+import ai.lum.common.ConfigUtils._
+import com.typesafe.config.{Config, ConfigFactory, ConfigValueFactory}
+import org.clulab.odin.{EventMention, Mention, RelationMention, TextBoundMention}
+import org.clulab.pdf2txt.Pdf2txt
+import org.clulab.pdf2txt.common.pdf.TextConverter
+import org.clulab.pdf2txt.languageModel.GigawordLanguageModel
+import org.clulab.pdf2txt.preprocessor._
+import org.clulab.utils.Logging
+import org.ml4ai.grounding.{GroundingCandidate, MiraEmbeddingsGrounder}
+import org.ml4ai.skema.text_reading.attachments.{GroundingAttachment, MentionLocationAttachment}
+import org.ml4ai.skema.text_reading.data.CosmosJsonDataLoader
+import org.ml4ai.skema.text_reading.serializer.AutomatesJSONSerializer
+
+import scala.collection.mutable.ArrayBuffer
+
+
+case class CosmosTextReadingPipeline() extends Logging {
+
+  /**
+    * Assigns grounding elements to a mention
+    * @param m mention to ground
+    * @return instance of the mention with grounding attachment if applicable
+    */
+  def groundMention(m: Mention): Mention = m match {
+    case tbm: TextBoundMention => {
+      val isGrounded =
```

Comment:
I'm not sure of this logic.  It seems to be the same as
```scala
val isGrounded = tbm.attachments.exists(!_.isInstanceOf[GroundingAttachment])
```
but that seems strange.